src/cluster.c

```diff
@@ -852,176 +689,306 @@ static dict *parse_cluster_slots(valkeyClusterContext *cc, valkeyReply *reply) {
     return NULL;
 }
 
-/**
- * Parse the "cluster nodes" command reply to nodes dict.
- */
-static dict *parse_cluster_nodes(valkeyClusterContext *cc, valkeyReply *reply) {
-    int ret;
-    dict *nodes = NULL;
-    dict *nodes_name = NULL;
-    valkeyClusterNode *master, *slave;
-    cluster_slot *slot;
-    char *pos, *start, *end, *line_start, *line_end;
-    char *role;
-    int role_len;
-    int slot_start, slot_end, slot_ranges_found = 0;
-    sds *part = NULL, *slot_start_end = NULL;
-    int count_part = 0, count_slot_start_end = 0;
-    int k;
-    int len;
-
-    if (reply->type != VALKEY_REPLY_STRING) {
-        valkeyClusterSetError(cc, VALKEY_ERR_OTHER, "Unexpected reply type");
-        goto error;
+/* Store a parsed replica node in given dict using the primary_id as key.
+ * Additional replicas for a primary are stored within the first replica. */
+static int store_replica_node(dict **replicas, char *primary_id, valkeyClusterNode *node) {
+    if (*replicas == NULL) {
+        *replicas = dictCreate(&clusterNodesDictType, NULL);
```

Comment:
This is a slightly odd API, but I guess there's some point in avoiding creating the dict if it isn't needed.... (Create it before we call this function doesn't make sense?)

Can we call it `replicas_ref` or something that indicates it's not the replicas dict itself but a reference or pointer to it?

Reply:
Changed it. Now creates the dict before calling this function.